### PR TITLE
Add engagement phases widget

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/885

*Description of changes:*
Add the engagement phases widget to engagement form

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/migrations/versions/2b12a6cd987a_add_phases_widget_type.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def upgrade():
+    widget_type_table = sa.table('widget_type',
+        sa.Column('id', sa.Integer),
+        sa.Column('name', sa.String),
+        sa.Column('description', sa.String))
+
+    op.bulk_insert(widget_type_table, [
+        {'id': 3, 'name': 'Phases', 'description': 'Displays information about the engagement phase'}
```
Comment (tiagograf-AOT) [inline met-api/migrations/versions/2b12a6cd987a_add_phases_widget_type.py:26]: I'm keeping it generic but we might want to be specific for this one, not sure yet

File: met-web/src/components/engagement/form/EngagementWidgets/Phases/PhasesOptionCard.tsx [hunk 2]
```diff
@@ -1,44 +1,52 @@
 import React, { useContext, useState } from 'react';
```
Comment (tiagograf-AOT) [inline met-web/src/components/engagement/form/EngagementWidgets/Phases/PhasesOptionCard.tsx:1]: I deleted a duplicated file and added a new one for phases, I'm not sure why it thinks I moved it

**Review by djnunez-aot — APPROVED** [review]

File: met-api/migrations/versions/7faee53e6759_alter_submission_add_comment_status.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def upgrade():
 
 def downgrade():
     conn = op.get_bind()
-    op.drop_constraint(None, 'submission', type_='foreignkey')
+    op.drop_constraint('submission_comment_status_id_fkey', 'submission', type_='foreignkey')
```
Comment (tiagograf-AOT) [inline met-api/migrations/versions/7faee53e6759_alter_submission_add_comment_status.py:43]: this fixes the downgrade migration, i'm piggying this

**Review by jadmsaadaot — APPROVED** [review]

**Comment by jadmsaadaot** [comment]
awesome

**Pull request merged**